spring-kafka/src/main/java/org/springframework/kafka/annotation/EnableRetryTopic.java

```diff
@@ -0,0 +1,91 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.annotation;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.kafka.config.RetryTopicConfigurationSupport;
+
+/**
+ * Enables the non-blocking topic-based delayed retries feature. To be used in
+ * {@link Configuration Configuration} classes as follows:
+ * <pre class="code">
+ *
+ * &#064;EnableRetryTopic
+ * &#064;Configuration
+ * public class AppConfig {
+ * }
+ *
+ * &#064;Component
+ * public class MyListener {
+ *
+ *     &#064;RetryableTopic(fixedDelayTopicStrategy = FixedDelayStrategy.SINGLE_TOPIC, backoff = @Backoff(4000))
+ *     &#064;KafkaListener(topics =  "myTopic")
+ * 	   public void listen(String message, @Header(KafkaHeaders.RECEIVED_TOPIC) String receivedTopic) {
+ *	       logger.info("Message {} received in topic {} ", message, receivedTopic);
+ *     }
+ *
+ *     &#064;DltHandler
+ *     public void dltHandler(Object message, @Header(KafkaHeaders.RECEIVED_TOPIC) String receivedTopic) {
+ *	       logger.info("Message {} received in dlt handler at topic {} ", message, receivedTopic);
+ *     }
+ * </pre>
+ *
+ * To configure the feature's components, extend the {@link RetryTopicConfigurationSupport}
+ * class and override the appropriate methods. Then import the subclass using the
+ * {@link Import @Import} annotation on a {@link Configuration @Configuration} class,
+ * such as:
+ *
+ * <pre class="code">
+ *
+ * &#064;Configuration
+ * &#064;EnableKafka
+ * &#064;Import(MyRetryTopicConfigurationSupport.class)
+ * public class AppConfig {
+ * }
+ *
+ * public static class MyRetryTopicConfigurationSupport extends RetryTopicConfigurationSupport {
+ *
+ * 		&#064;Override
+ * 		protected void configureBlockingRetries(BlockingRetriesConfigurer blockingRetries) {
+ * 			blockingRetries
+ * 				.retryOn(ShouldRetryOnlyBlockingException.class, ShouldRetryViaBothException.class)
+ * 				.backOff(new FixedBackOff(50, 3));
+ * 		}
+ *
+ * 		&#064;Override
+ * 		protected void configureNonBlockingRetries(NonBlockingRetriesConfigurer nonBlockingRetries) {
+ * 			nonBlockingRetries
+ * 				.addToFatalExceptions(ShouldSkipBothRetriesException.class);
+ * } *
+ * </pre>
+ *
+ * @author Tomaz Fernandes
+ * @since 2.9
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+@Documented
+@Import(RetryTopicConfigurationSupport.class)
+public @interface EnableRetryTopic {
```

Comment:
Perhaps `@EnableKafkaRetryTopic`? There are many `@Enable*` annotations across the Spring portfolio.

Reply:
Sure, makes sense. @artembilan and I discussed somewhere about adding this `Support` class or this annotation directly to `@EnableKafka` in 3.0, and thus having this bootstrapping as a framework default. So the purpose of this annotation would be an intermediate step for 2.9.x.

But I think we can have this single `@EnableKafkaRetryTopic ` annotation for now and import the `Support` class directly in the `@EnableKafka` in the future.

WDYT?